### PR TITLE
CBL-2504: Add submodule verification script

### PR DESCRIPTION
**Pull request by jayahariv**
* Reused the script from lite core 
* Related PR - https://github.com/couchbase/couchbase-lite-core/pull/1357

### CODE REVIEW

**Comment by jayahariv** [comment]
PR validation failure is not related to this! I will check it separately.

**Comment by pasin** [comment]
Can you provide the info or steps that the `check_deps.py` is doing?

**Comment by borrrden** [comment]
Ideally MYUtilities will be migrated to couchbasedeps and appropriately branched or tagged

**Comment by jayahariv** [comment]
Once its moved, then we could use the same script in LiteCore with a path argument or working directory path.

**Review by pasin — COMMENTED** [review]
Will there be update to the PR?

**Comment by jayahariv** [comment]
yes, there will be

**Comment by jayahariv** [comment]
Once the LC PR is merged, I will resubmit again!

File: .github/workflows/check_deps.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    - uses: actions/checkout@v2
+      with:
+        submodules: true # Only need one level for this check
+        fetch-depth: 0 # Pull all history or else we won't get branch info
+
+    - name: Check Submodules
+      working-directory: ${{github.workspace}}
+      run: |
+        pip install termcolor
+        ./vendor/couchbase-lite-core/build_cmake/scripts/check_deps.py -branch $GITHUB_BASE_REF -path ${{github.workspace}}/
```
Comment (jayahariv) [inline .github/workflows/check_deps.yml:29]:
```suggestion
        ./vendor/couchbase-lite-core/build_cmake/scripts/check_deps.py -branch $GITHUB_BASE_REF -path ${{github.workspace}}
```

File: .github/workflows/check_deps.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    - uses: actions/checkout@v2
+      with:
+        submodules: true # Only need one level for this check
+        fetch-depth: 0 # Pull all history or else we won't get branch info
+
+    - name: Check Submodules
+      working-directory: ${{github.workspace}}
+      run: |
+        pip install termcolor
+        ./vendor/couchbase-lite-core/build_cmake/scripts/check_deps.py -branch $GITHUB_BASE_REF -path ${{github.workspace}}
```
Comment (jayahariv) [inline .github/workflows/check_deps.yml:29]:
```suggestion
        ./vendor/couchbase-lite-core/build_cmake/scripts/check_deps.py $GITHUB_BASE_REF ${{github.workspace}}
```

File: .github/workflows/check_deps.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+# Copyright 2022-Present Couchbase, Inc.
```
Comment (jayahariv) [inline .github/workflows/check_deps.yml:1]:
```suggestion
# Copyright 2022-Present Couchbase, Inc..
```

File: .github/workflows/check_deps.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
+# Copyright 2022-Present Couchbase, Inc..
```
Comment (jayahariv) [inline .github/workflows/check_deps.yml:1]:
```suggestion
# Copyright 2022-Present Couchbase, Inc.
```

**Review by pasin — APPROVED** [review]

**Pull request merged**